### PR TITLE
Fixes and improvements to cloud sources

### PR DESCRIPTION
**Pull request by ramonhpr**
This PR makes fixes and improvements to cloud sources.

### CODE REVIEW

File: src/amqp.c [hunk 1]
```diff
@@ -216,6 +216,14 @@ static int start_connection(struct settings *settings)
 	return 0;
 }
 
+/**
+ * amqp_declare_new_queue:
```
Comment (vitbaq) [inline src/amqp.c]: :eyes:

File: src/amqp.c [hunk 2]
```diff
@@ -216,6 +216,14 @@ static int start_connection(struct settings *settings)
...
+ * @name: queue name
+ *
+ * This function declare a durable queue in amqp.
```
Comment (vitbaq) [inline src/amqp.c]: Remove "This".

File: src/amqp.c [hunk 3]
```diff
@@ -256,6 +264,14 @@ int8_t amqp_publish_persistent_message(const char *exchange,
 	return rc;
 }
 
+/**
+ * amqp_declare_new_queue:
+ * @name: queue name
+ *
+ * This function declare a durable queue in amqp.
```
Comment (vitbaq) [inline src/amqp.c]: Remove "This", can be: "Declare a durable queue in amqp"

File: src/amqp.c [hunk 4]
```diff
@@ -282,6 +298,17 @@ amqp_bytes_t amqp_declare_new_queue(const char *name)
 	return queue;
 }
 
+/**
+ * amqp_set_queue_to_consume:
+ * @queue: queue declared
+ * @exchange: exchange to be declared
+ * @routing_key: routing key to bind
+ *
+ * This function declare a exchange and bind a routing key to a queue to be a
```
Comment (vitbaq) [inline src/amqp.c]: Remove "This. Can be "Declare..." or "Function to declare..."

File: src/amqp.c [hunk 5]
```diff
@@ -329,6 +356,16 @@ int amqp_set_queue_to_consume(amqp_bytes_t queue,
 	return 0;
 }
 
+/**
+ * amqp_set_read_cb:
+ * @on_read: callback to be called when receive some amqp message
+ * @user_data: user data provided to callback
+ *
+ * This function is responsible to set the callback handler when receive some
```
Comment (vitbaq) [inline src/amqp.c]: Same here

File: src/cloud.c [hunk 6]
```diff
@@ -154,11 +151,20 @@ static bool on_cloud_receive_message(const char *exchange,
...
-done:
 	json_object_put(jso);
 	return consumed;
 }
 
+/**
+ * cloud_unregister_device:
+ * @id: device id
+ *
+ * This function is responsible to request cloud to remove a device.
```
Comment (vitbaq) [inline src/cloud.c]: Same here, remove "This"

File: src/cloud.c [hunk 7]
```diff
@@ -179,6 +185,18 @@ int cloud_unregister_device(const char *id)
...
 
+/**
+ * cloud_publish_data:
+ * @id: device id
+ * @sensor_id: schema sensor id
+ * @value_type: schema value type defined in KNoT protocol
+ * @value: value to be sent
+ * @kval_len: length of @value
+ *
+ * This function is responsible to send data to cloud.
```
Comment (vitbaq) [inline src/cloud.c]: Same here

File: src/cloud.c [hunk 8]
```diff
@@ -200,7 +218,19 @@ int cloud_publish_data(const char *id, uint8_t sensor_id, uint8_t value_type,
...
 
-int cloud_set_cbs(cloud_downstream_cb_t on_update,
+/**
+ * cloud_set_read_handlers:
+ * @on_update: callback to be called when receive an update message
+ * @on_request: callback to be called when receive an request message
+ * @on_removed: callback to be called when device is removed
+ * @user_data: user data provided to callbacks
+ *
+ * This function is responsible to set callback handlers when receive cloud
```
Comment (vitbaq) [inline src/cloud.c]: Same here

File: src/amqp.c (hunk 2 above)
Comment (ramonhpr) [reply src/amqp.c]: Done

File: src/amqp.c (hunk 1 above)
Comment (ramonhpr) [reply src/amqp.c]: Done

File: src/amqp.c (hunk 3 above)
Comment (ramonhpr) [reply src/amqp.c]: Done

File: src/amqp.c (hunk 4 above)
Comment (ramonhpr) [reply src/amqp.c]: Done

File: src/amqp.c (hunk 5 above)
Comment (ramonhpr) [reply src/amqp.c]: Done

File: src/cloud.c (hunk 6 above)
Comment (ramonhpr) [reply src/cloud.c]: Done

File: src/cloud.c (hunk 7 above)
Comment (ramonhpr) [reply src/cloud.c]: Done

File: src/cloud.c (hunk 8 above)
Comment (ramonhpr) [reply src/cloud.c]: Done

File: src/cloud.c [hunk 9]
```diff
@@ -154,7 +154,6 @@ static bool on_cloud_receive_message(const char *exchange,
 		break;
```
Comment (alla3) [inline src/cloud.c]: Please, remove the break.

Comment (ramonhpr) [reply src/cloud.c]: Done

File: src/cloud.c [hunk 10]
```diff
@@ -123,15 +123,12 @@ static bool on_cloud_receive_message(const char *exchange,
 	case UPDATE_EVT:
 		cb_handler.update_cb = evt_handler->cb;
 		id = get_id_from_json_obj(jso);
-		if (!id) {
+		list = parser_update_to_list(jso);
+		if (!id || !list) {
```
Comment (alla3) [inline src/cloud.c]: If your id is NULL and list is not null you do not destroy the list.

**Pull request merged**